ngraph/frontend/generic/include/frontend_manager/frontend_manager.hpp

```diff
@@ -0,0 +1,454 @@
+
+            /// \brief Set name for a particular dimension of a place (e.g. batch dimension)
+            virtual void set_name_for_dimension(Place::Ptr place,
+                                                size_t shapeDimIndex,
+                                                const std::string& dimName);
+
+            ///// Topology Editing  /////
+
+            /// \brief Cut immediately before this place and assign this place as new input; prune
+            /// all nodes that don't contribute to any output.
+            virtual void cut_and_add_new_input(Place::Ptr place,
+                                               const std::string& newNameOptional = "");
+
+            /// \brief Cut immediately after this place and assign this place as new output; prune
+            /// all nodes that don't contribute to any output.
+            virtual void cut_and_add_new_output(Place::Ptr place,
+                                                const std::string& newNameOptional = "");
+
+            /// \brief Assign this place as new output or add necessary nodes to represent a new
+            /// output.
+            ///
+            /// \param place Anchor point to add an output
+            /// \return new output place, may be the same as a given place
+            virtual Place::Ptr add_output(Place::Ptr place);
+
+            /// \brief Removes any sinks directly attached to this place with all inbound data flow
+            /// if it is not required by any other output.
+            virtual void remove_output(Place::Ptr place);
+
+            /// \brief Removes an input place and all data flow that depends on it.
+            // TODO: remove it as something not practically useful in the API?
+            virtual void remove_input(Place::Ptr place);
+
+            /// \brief Replaces all existing outputs with new ones removing all data flow that is
+            /// not required for new outputs.
+            ///
+            /// \param outputs Vector with places that will become new outputs; may intersect
+            /// existing outputs.
+            virtual void override_all_outputs(const std::vector<Place::Ptr>& outputs);
+
+            /// \brief Modifies the graph to use new inputs instead of existing ones. New inputs
+            /// should completely satisfy all existing outputs.
+            virtual void override_all_inputs(const std::vector<Place::Ptr>& inputs);
+
+            /// \brief Leaves only subgraph that are defined by new inputs and new outputs.
+            virtual void extract_subgraph(const std::vector<Place::Ptr>& inputs,
+                                          const std::vector<Place::Ptr>& outputs);
+
+            ///// Setting tensor properties  /////
+
+            /// Sets shape that would be used by default for this place; place should be uniquely
+            /// refer to some data.
+            // TODO: define clearly which scenario requires it -- currently it should satisfy
```

Comment:
If we don't know the scenario for this, maybe can we exclude it from API?

Reply:
Agree, I would like to keep it in PoC, but not in master. Should be aligned with IR v10 extension when we are going to distinguish partial shape and default (static) shape.

Reply:
Removed from master PR